gradio/routes.py

```diff
@@ -292,17 +295,16 @@ async def file(path_or_url: str, request: fastapi.Request):
                 )
             abs_path = str(utils.abspath(path_or_url))
             in_app_dir = utils.abspath(app.cwd) in utils.abspath(path_or_url).parents
-            created_by_app = str(utils.abspath(path_or_url)) in set().union(
-                *blocks.temp_file_sets
-            )
+            created_by_app = abs_path in set().union(*blocks.temp_file_sets)
             in_file_dir = any(
                 (
                     utils.abspath(dir) in utils.abspath(path_or_url).parents
                     for dir in blocks.file_directories
                 )
             )
+            was_uploaded = abs_path.startswith(app.uploaded_file_dir)
```

Comment:
Is `startswith()` robust? I think using the `in` operator with two `Path`s is more robust

Reply:
No, it's the opposite - the `in` keyword is not robust. For example, if the upload dir is "a/b", and a user tries to access file "c/d/a/b", the `in` keyword will allow access to this. This is not secure. We should modify the other lines to use `startswith` as well - I can do this is in this PR

Reply:
Not `in` with two strings, but `in` with two `pathlib.Path` objects. See here: https://stackoverflow.com/a/34236245/5209347

Reply:
oh yep you're right